src/main/kotlin/lotto/domain/LottoStatCalculator.kt

```diff
@@ -0,0 +1,25 @@
+package lotto.domain
+
+class LottoStatCalculator(private val winningLotto: WinningLotto) {
+
+    fun getStat(lottoList: List<Lotto>): LottoStatResult {
+        val result = LottoStatResult().apply {
+            lottoList.forEach {
+                this.addCount(Rank.valueOf(compareWithWinningLotto(it), getIsMatchBonusNumber(it)))
+            }
+        }
+
+        return result
+    }
```

Comment:
p3; 코드 스타일이라 조심스럽지만, 어떤 목적임은 알겠지만 addCount는 불필요한 메서드가 아닐까 싶어요.

```suggestion
    fun getStat(lottery: Lottery): LottoStatResult = LottoStatResult(
        lottoStat = lottery.groupingBy { 
            Rank.valueOf(
                countOfMatch = compareWithWinningLotto(lotto = it),
                matchBonus = getIsMatchBonusNumber(lotto = it),
            ) 
        }.eachCount()
    )
```

이와 같이 컬렉션 함수로 처리하는 부분을 LottoStatResult에서 *Lottery를 생성자의 인자로 받아 처리할 수 있을 거 같아요.
위의 로직은 Calculator가 결과에 대한 연산을 책임지고 있으니, 예시로 참고해 주시고 적절한 객체에게 역할을 위임하는 것도 생각해 보시면 좋을 거 같아요.

*Lottery 예시로 만든 일급 컬렉션입니다.